array/array.go

```diff
@@ -0,0 +1,131 @@
+package array
+
+import (
+	"errors"
+)
+
+type Array struct {
+	arr  []int
+	size int
+}
+
+func NewArray(capacity int) *Array {
+	return &Array{
+		arr: make([]int, capacity),
+	}
+}
+
+func (a *Array) Size() int {
+	return a.size
+}
+
+func (a *Array) Capacity() int {
+	return len(a.arr)
+}
+
+func (a *Array) IsEmpty() bool {
+	return a.Size() == 0
+}
+
+func (a *Array) ValueAt(index int) (int, error) {
+	if index < 0 || index >= a.Size() {
```

Comment:
In this function, range is from 0 to size but in another function range is from 0 to capacity. It will make user confuse when using.